src/adagrams.js

```diff
@@ -1,15 +1,165 @@
-export const drawLetters = () => {
-  // Implement this method for wave 1
-};
+class Adagrams{
+  LETTER_POOL = {
+    A: 9,
+    B: 2,
+    C: 2,
+    D: 4,
+    E: 12,
+    F: 2,
+    G: 3,
+    H: 2,
+    I: 9,
+    J: 1,
+    K: 1,
+    L: 4,
+    M: 2,
+    N: 6,
+    O: 8,
+    P: 2,
+    Q: 1,
+    R: 6,
+    S: 4,
+    T: 6,
+    U: 4,
+    V: 2,
+    W: 2,
+    X: 1,
+    Y: 2,
+    Z: 1,
+  };
 
-export const usesAvailableLetters = (input, lettersInHand) => {
-  // Implement this method for wave 2
-};
+  LETTER_SCORE = {
+    A: 1,
+    B: 3,
+    C: 3,
+    D: 2,
+    E: 1,
+    F: 4,
+    G: 2,
+    H: 4,
+    I: 1,
+    J: 8,
+    K: 5,
+    L: 1,
+    M: 3,
+    N: 1,
+    O: 1,
+    P: 3,
+    Q: 10,
+    R: 1,
+    S: 1,
+    T: 1,
+    U: 1,
+    V: 4,
+    W: 4,
+    X: 8,
+    Y: 4,
+    Z: 10,
+  };
+  
+  constructor() {
+    this.input = "";
+    this.word = "";
+    this.highest = {
+      word: "",
+      score: 0
+    };
+    
+    this.lettersInHand = new Array();
+
+  };
+
+  drawLetters() {
+    while (this.lettersInHand.length < 10) {
+      let letter = String.fromCharCode(Math.floor(Math.random() * 26) + 65);
+      
+      if (this.LETTER_POOL[letter] > 0) {
+          this.LETTER_POOL[letter] -= 1;
+          this.lettersInHand.push(letter);
+        }
+      letter = String.fromCharCode(Math.floor(Math.random() * 26) + 65);
+        
+    };
+    return this.lettersInHand;
+  };
+
+  isValid(word) {
+    if(word.length < 1) return false;
+    
+    for(let i=0; i < word.length; ++i){
+      
+      if (!(/^[A-Z]$/.test(word[i])) ){
+        return false;
+      }
+    }
+    return true;
+  };
+
+  usesAvailableLetters(input, lettersInHand) {
+    this.input = input.toUpperCase();
+    if (!this.isValid(this.input)){
+      return false;
+    }
+    this.lettersInHand = lettersInHand;
+  
+    let letterFreq = {};
+    for (let i=0; i < this.lettersInHand.length; i++) {
+      let letter = this.lettersInHand[i]
+      letterFreq[letter] = (letterFreq[letter] || 0) +1 ;
+    };
+    
+    for (let i=0; i < this.input.length; i++) {
+      let letter = this.input[i];
+      if ((letter in letterFreq)&&(letterFreq[letter] > 0)) {
+        letterFreq[letter] -= 1;
+      } else {
+        return false;
+      }
+    };
+    return true;
+  };
+  
+  scoreWord(word) {
+    this.word = word.toUpperCase();
+    if (!this.isValid(this.word)){
+      return 0;
+    }
+    let point = 0;
+    
+    for (let i=0; i < this.word.length; ++i){
+      point += this.LETTER_SCORE[this.word[i]];
+    }
```

Comment:
This could have an unexpected result if the string `word` has characters that aren't in `LETTER_SCORE `. If we have an input like `"ABC DEF"` then `point` will hold `NaN` (Not a Number) after the loop. If we wanted to skip non-alphabetic characters or characters that aren't in `LETTER_SCORE `, how could we do that?